deepspeed/profiling/flops_profiler/README.md

```diff
@@ -35,216 +38,324 @@ Top 3 modules in flops at depth 2 are {'Conv2d': '421.91 MMACs', 'Linear': '11.1
+    58.63 M, 95.96% Params, 15.01 GMACs, 8.19% MACs, 1.7 ms, 27.49% time, 1.8e+01 TFLOPS,
+    (0): Dropout(0, 0.00% Params, 0 MACs, 0.00% MACs, 14.31 us, 0.23% time, 0.0 TFLOPS, p=0.5, inplace=False)
+    (1): Linear(37.75 M, 61.79% Params, 9.66 GMACs, 5.28% MACs, 83.21 us, 1.34% time, 2.3e+02 TFLOPS, in_features=9216, out_features=4096, bias=True)
+    (2): ReLU(0, 0.00% Params, 1.05 MMACs, 0.00% MACs, 25.03 us, 0.40% time, 0.084 TFLOPS, inplace=True)
+    (3): Dropout(0, 0.00% Params, 0 MACs, 0.00% MACs, 10.49 us, 0.17% time, 0.0 TFLOPS, p=0.5, inplace=False)
+    (4): Linear(16.78 M, 27.46% Params, 4.29 GMACs, 2.34% MACs, 67.23 us, 1.09% time, 1.3e+02 TFLOPS, in_features=4096, out_features=4096, bias=True)
+    (5): ReLU(0, 0.00% Params, 1.05 MMACs, 0.00% MACs, 22.17 us, 0.36% time, 0.095 TFLOPS, inplace=True)
+    (6): Linear(4.1 M, 6.71% Params, 1.05 GMACs, 0.57% MACs, 1.44 ms, 23.33% time, 1.5 TFLOPS, in_features=4096, out_features=1000, bias=True)
+  )
+)
+Top 3 modules in flops at depth 2 are {'Conv2d': '167.95 GMACs', 'Linear': '15.01 GMACs', 'ReLU': '126.26 MMACs'}
+Top 3 modules in params at depth 2 are {'Linear': '58.63 M', 'Conv2d': '2.47 M', 'ReLU': '0'}
+Top 3 modules in time at depth 2 are {'Conv2d': '4.05 ms', 'Linear': '1.59 ms', 'ReLU': '223.4 us'}
+Batch size:                     256
+Number of MACs:                 183.18 GMACs
+Number of parameters:           61.1 M
 ```
 
-##### Bert model example:
+#### Example: Bert
 
 ```python
 from functools import partial
-
 import torch
 from transformers import BertForSequenceClassification, BertTokenizer
-
 from deepspeed.profiling.flops_profiler import get_model_profile
 
 
 def bert_input_constructor(input_shape, tokenizer):
-    inp_seq = ""
-    for _ in range(input_shape[1] - 2):  # there are two special tokens [CLS] and [SEP]
-        inp_seq += tokenizer.pad_token  # let's use pad token to form a fake
-    # sequence for subsequent flops calculation
-
-    inputs = tokenizer([inp_seq] * input_shape[0],
+    fake_seq = ""
+    for _ in range(input_shape[1] - 2):  # ignore the two special tokens [CLS] and [SEP]
+        fake_seq += tokenizer.pad_token
+    input_tensor = tokenizer([fake_seq] * input_shape[0],
                        padding=True,
                        truncation=True,
                        return_tensors="pt")
-    labels = torch.tensor([1] * input_shape[0])
-    # Batch size input_shape[0], sequence length input_shape[128]
-    inputs = dict(inputs)
-    inputs.update({"labels": labels})
-    return inputs
+    return input_tensor
 
 
-if __name__ == '__main__':
-    bert_tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
+with torch.cuda.device(0):
+    tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
     model = BertForSequenceClassification.from_pretrained('bert-base-uncased')
-    macs, params, steps = get_model_profile(
+    batch_size = 5
+    seq_len = 128
+    macs, params = get_model_profile(
```

Comment:
It would be good to how how the model is called without the profiling so that the input_res and input_consturctors are clearer. Maybe have:
if profile:
macs, params = get_model_profile
else:
output/loss = model(....)

Reply:
Made changes as suggested